synapse/lib/storm.py

```diff
@@ -918,3 +925,79 @@ def _stormOperDelTag(self, query, oper):
 
         for tag in tags:
             [ core.delTufoTag(node,tag) for node in nodes ]
+
+    def _stormOperJoinTags(self, query, oper):
+        args = oper[1].get('args')
+        opts = dict(oper[1].get('kwlist'))
+
+        nodes = query.data()
+
+        keep_nodes = opts.get('keep_nodes', False)
+        limt = LimitHelp(opts.get('limit'))
+
+        forms = {arg for arg in args}
+
+        tags = {tag for node in nodes for tag in s_tufo.tags(node, leaf=True)}
+
+        core = self.getStormCore()
+
```

Comment:
getFormDefs() actually requires the form to be known ahead of time; but getModelDict will return a dictionary containing a list of forms which will suffice.